sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2342,6 +2342,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_CENTRAL_MOMENT_AGG =
+    buildConf("spark.sql.legacy.centralMomentAgg")
+      .internal()
+      .doc("When set to true, central moment aggregation will return Double.NaN " +
+        "if divide by zero occurred during calculation. " +
+        "Otherwise, it will return null")
```

Comment:
Can we describe the Spark version of this legacy behavior? E.g., In what versions it returns NaN.

Reply:
sure. how about adding "before version 3.1.0, it returns NaN by default."